specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-12-01/Marketplace.json

```diff
@@ -0,0 +1,2894 @@
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CollectionsToSubscriptionsMappingPayload": {
+      "description": "The subscriptions list to get the related collections",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/CollectionsToSubscriptionsMappingProperties",
+          "description": "Subscriptions ids list"
+        }
+      }
+    },
+    "CollectionsToSubscriptionsMappingProperties": {
+      "description": "The subscriptions list to get the related collections",
+      "type": "object",
+      "properties": {
+        "subscriptionIds": {
+          "description": "Subscriptions ids list",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CollectionsToSubscriptionsMappingResponse": {
+      "description": "A map of collections subscriptions details",
+      "type": "object",
+      "properties": {
+        "details": {
+          "$ref": "#/definitions/CollectionsToSubscriptionsMappingResponseProperties",
+          "description": "The map of collections subscriptions"
+        }
+      }
+    },
+    "CollectionsToSubscriptionsMappingResponseProperties": {
+      "description": "Collections - subscriptions mapping details, map key is collectionId. In case subscriptions list is null or empty: no subscription from the list is related to that collection.",
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/collectionsSubscriptionsMappingDetails"
+      }
+    },
+    "collectionsSubscriptionsMappingDetails": {
+      "description": "Collection name and related subscriptions list",
+      "type": "object",
+      "properties": {
+        "collectionName": {
+          "description": "Collection name",
+          "type": "string"
+        },
+        "subscriptions": {
+          "description": "Subscriptions ids list",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "StopSellOffersPlansNotificationsList": {
+      "description": "List of stop sell offers and plans notifications.",
+      "type": "object",
+      "properties": {
+        "stopSellNotifications": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StopSellOffersPlansNotificationsListProperties"
+          }
+        }
+      }
+    },
+    "StopSellOffersPlansNotificationsListProperties": {
+      "description": "List of stop sell offers and plans notifications.",
+      "type": "object",
+      "properties": {
+        "offerId": {
+          "description": "Gets offer id",
+          "type": "string",
+          "readOnly": true
+        },
+        "displayName": {
+          "description": "Gets offer display name",
+          "type": "string",
+          "readOnly": true
+        },
+        "isEntire": {
```

Comment:
>isEntire [](http://example.com/codeflow?start=9&length=8)

ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use.

Wiki page: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review)
Best option is to choose a better property (e.g. stopSellStatus: { "Partial", "Full" }). This type of property is easier for customers to understand and use. Next best option if you really want a True/False value is to use a string enum, e.g. "fullyStopped": { "True", "False" }. Enums allow adding values later without requiring a new API version.

Reply:
IsEntire means if all the offer is stop sell or just plan(s)
We have this term in an existing API and we would like to stay with this term to be aligned with the previous API version with similar logic

Reply:
If it is necessary I can change the boolean to enum.

Reply:
It's not a requirement, just a recommendation. Please consider this design pattern going forward. ARM is trying to move away from APIs with boolean properties.